os/arch/arm/src/amebasmart/amebasmart_i2s.c

```diff
@@ -1663,7 +1663,12 @@ static void i2s_getdefaultconfig(struct amebasmart_i2s_s *priv)
 	}
 	priv->txenab = priv->config->txenab;
 	if (priv->config->txenab) {
+#if defined(CONFIG_AUDIO_NDP120)
```

Comment:
I think this macro is only in internal codebase. Is it ok to apply it in public?

Reply:
I am not 100% sure who should really answer this, however:
`CONFIG_AUDIO_NDP120` is already in the public codebase, here is a quick grep from a fresh clone of TizenRT repo done today (HEAD at commit 2901b73b196c9b49e9970bd4a76db6ae69788deb)
```
~/tmp/TizenRT (master ✔) grep -riIn CONFIG_AUDIO_NDP120
./os/drivers/audio/Make.defs:83:ifeq ($(CONFIG_AUDIO_NDP120),y)
./os/board/rtl8730e/src/Make.defs:75:ifeq ($(CONFIG_AUDIO_NDP120),y)
./os/board/rtl8730e/src/rtl8730e_ndp120.c:40:#ifdef CONFIG_AUDIO_NDP120
./os/board/rtl8730e/src/rtl8730e_boot.c:488:#ifdef CONFIG_AUDIO_NDP120
./os/include/tinyara/audio/ndp120.h:32:#ifdef CONFIG_AUDIO_NDP120
./os/include/tinyara/audio/ndp120.h:129:#endif							/* CONFIG_AUDIO_NDP120 */
./build/configs/rtl8730e/audio/defconfig:531:CONFIG_AUDIO_NDP120=y
~/tmp/TizenRT (master ✔)
```

Reply:
@edwakuwaku
Hello edward, I think you can provide some config regarding this.
Something like a 'CONFIG_AMEBASMART_TXCL'
Then we can set default value as 16 or 32, we will take 16 when we use ndp120.
How do you think?
@abhishek-samsung  Please let me know your opinion too.

Reply:
> I am not 100% sure who should really answer this, however: `CONFIG_AUDIO_NDP120` is already in the public codebase, here is a quick grep from a fresh clone of TizenRT repo done today (HEAD at commit [2901b73](https://github.com/Samsung/TizenRT/commit/2901b73b196c9b49e9970bd4a76db6ae69788deb))
> 
> ```
> ~/tmp/TizenRT (master ✔) grep -riIn CONFIG_AUDIO_NDP120
> ./os/drivers/audio/Make.defs:83:ifeq ($(CONFIG_AUDIO_NDP120),y)
> ./os/board/rtl8730e/src/Make.defs:75:ifeq ($(CONFIG_AUDIO_NDP120),y)
> ./os/board/rtl8730e/src/rtl8730e_ndp120.c:40:#ifdef CONFIG_AUDIO_NDP120
> ./os/board/rtl8730e/src/rtl8730e_boot.c:488:#ifdef CONFIG_AUDIO_NDP120
> ./os/include/tinyara/audio/ndp120.h:32:#ifdef CONFIG_AUDIO_NDP120
> ./os/include/tinyara/audio/ndp120.h:129:#endif							/* CONFIG_AUDIO_NDP120 */
> ./build/configs/rtl8730e/audio/defconfig:531:CONFIG_AUDIO_NDP120=y
> ~/tmp/TizenRT (master ✔)
> ```

Hi @cvejlbo , yeap it was merged recently, thank you for reminding~

Reply:
Hello Taejun님, do you mean something like this? https://github.com/Samsung/TizenRT/pull/6348

Reply:
> @Taejun-Kwon Hello Taejun님, do you mean something like this? #6348

Hello,
Yes. I mean this.
BTW is channel length of i2s can be changed dynamically in the middle of sending?

Reply:
Hello @Taejun-Kwon , I think this should be okay. However, the new amp is not verfied with datawidth as 16, so we will have to check with it. @cvejlbo , can ndp also work with 3MHz instead of 1.5MHz?

Reply:
Hello~ The channel length cannot be changed dynamically during runtime. As it will affect the BLCK and result in error during data transfer.

Reply:
> Hello @Taejun-Kwon , I think this should be okay. However, the new amp is not verfied with datawidth as 16, so we will have to check with it. @cvejlbo , can ndp also work with 3MHz instead of 1.5MHz?

In general, the NDP can work with 3.072MHz PDM clock as well, however the DMIC will definitely have higher power consumption, and it may also have a different sensitivity given that clock rate.
We'd also need to check/review the impact on NDP side (adjust for sensitivity levels, etc) with the given DMIC at 3.072MHz if the PDM clock rate is changed.

Reply:
> @cvejlbo , thanks for explaining. Let us stick with 1.5MHz. This file change can be removed as we will merge #6348 for this.

@abhishek-samsung, Sounds good. In order to not break anything, I guess it's OK if I wait with reverting the changes in `amebasmart_i2s.c` until #6348 is merged?

Reply:
Sure, we will merge #6348 asap.

Reply:
@cvejlbo , we merged https://github.com/Samsung/TizenRT/pull/6348, can you please update the PR to include the new changes?

Reply:
> @cvejlbo , we merged #6348, can you please update the PR to include the new changes?

Done